### ISSUE TITLE
Question about PPO recurrent policy

### ISSUE THREAD

**Issue by gliese581gg** [ikostrikov/pytorch-a2c-ppo-acktr#82]
Hi!

In your codes,

It looks like the gru states are computed with 'torch.no_grad()'.

Then PPO agent unrolls only one step from each of stored gru state when updates its parameters.

Is it okay to not compute the gradient of Loss_t regards to h_(t-1)?

**Comment by ikostrikov** [comment]
Hi.

Yes, the states are computed with torch.no_grad but then it recomputes all intermediate inside a computational graph:

https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/blob/master/model.py#L124
